aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SyncProcessor.java

```diff
@@ -113,9 +114,18 @@ Completable hydrate() {
                         .flatMapCompletable(merger::merge)
                     )
                     .andThen(syncTimeRegistry.saveLastSyncTime(modelClass, SyncTime.now()))
-                    .doOnError(failureToSync ->
-                        LOG.warn("Initial cloud sync failed.", failureToSync)
-                    )
+                    .doOnError(failureToSync -> {
+                        LOG.warn("Initial cloud sync failed.", failureToSync);
+                        DataStoreErrorHandler dataStoreErrorHandler =
+                            dataStoreConfigurationProvider.getConfiguration().getDataStoreErrorHandler();
```

Comment:
Can `getConfiguration()` ever return `null`? Is there a need to guard against that scenario, here, to prevent a `NullPointerException`?

Reply:
I don't think so. That parameter comes from the [plugin's constructor](https://github.com/aws-amplify/amplify-android/blob/074fae16941849ecac7b83cfe3bb04b50bcb4a54/aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java#L101) and passes the plugin configuration that's initialized [here](https://github.com/aws-amplify/amplify-android/blob/074fae16941849ecac7b83cfe3bb04b50bcb4a54/aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java#L174). Since the orchestrator is only started after the plugin is configured, I think we're safe here.